### PR TITLE
[IOTDB-2610] Add metrics for compation module

### PR DESCRIPTION
**Pull request by THUMarkLau**
See [IOTDB-2610](https://issues.apache.org/jira/browse/IOTDB-2610).
This PR add some metrics for compaction module, including read rate, write rate, task info etc.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionMetricsManager.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+import org.apache.iotdb.db.engine.compaction.inner.AbstractInnerSpaceCompactionTask;
+import org.apache.iotdb.db.engine.compaction.task.AbstractCompactionTask;
+import org.apache.iotdb.db.service.metrics.Metric;
+import org.apache.iotdb.db.service.metrics.MetricsService;
+import org.apache.iotdb.db.service.metrics.Tag;
+import org.apache.iotdb.metrics.utils.MetricLevel;
+
+import java.util.concurrent.TimeUnit;
+
+public class CompactionMetricsManager {
```
Comment (stormbroken) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionMetricsManager.java:34]: Maybe you can check metric module whether is used in this class, and not use `if (MetricConfigDescriptor.getInstance().getMetricConfig().getEnableMetric())` in other module

**Review by stormbroken — CHANGES_REQUESTED** [review]
I think you can check metric module whether start in CompactionMetricsManager

**Comment by stormbroken** [comment]
Excellent Job! I think you can consider whether metrics depend on their status before(mainly for hot reload).
Besides, I suggest that you can add a method in `CompactionMetricManager` to init all defined metrics.

**Comment by stormbroken** [comment]
Oh, I think you need to update doc at the same time.
1. docs/UserGuide/Maintenance-Tools/Metric-Tool.md
2. docs/zh/UserGuide/Maintenance-Tools/Metric-Tool.md

**Review by SpriCoder — APPROVED** [review]
LGTM! Only one minor problem: please update documentation.

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionMetricsManager.java (hunk 1 above)
Comment (THUMarkLau) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionMetricsManager.java:35]: resolve

**Comment by SteveYurongSu** [comment]
Should it be cherry-picked to the rel/0.13? @qiaojialin

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/constant/CompactionTaskStatus.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.engine.compaction.constant;
+
+public enum CompactionTaskStatus {
+  ADD_TO_QUEUE,
+  POLL_FROM_QUEUE,
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/constant/CompactionTaskStatus.java:23]: This looks like action but not status

**Pull request merged**